app/services/notification_handler.rb

```diff
@@ -0,0 +1,53 @@
+class NotificationHandler
+  attr_reader :params
+  def initialize(params:)
+    @params = params
+  end
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def call
+    begin
```

Comment:
Do we need to wrap this in a transaction to think about what happens if only the second operation fails?

Reply:
I thought about that but I think if the first one fails we get the error report and if the second one fails then we get an error report but still have the notification that we can use to fix it. I decided we probably didn't want the first db call to not be committed because of the second one failing.

Reply:
I suppose if we're thinking of `Notification` as a log of all notifications then it is important this gets committed to the db. I imagine we'll add similar sort of Icinga alerts to this new system, so perhaps a mismatched number of emails and notifications could be something we proactively check for.

Reply:
Yeah... That would be good